recipes/dd-agent.rb

```diff
@@ -140,6 +141,11 @@ def template_vars
 
 service 'datadog-agent' do
   service_name service_name
+  if is_amazon_linux_2023
+    status_command "systemctl status #{service_name}"
+    start_command "systemctl start #{service_name}"
+    stop_command "systemctl stop #{service_name}"
+  end
```

Comment:
I'm surprised that this would be needed specifically for Amazon Linux 2023. Is Chef not able to resolve the correct service provider?

Reply:
On the test I ran yesterday I was facing this
```
    ================================================================================
    Error executing action `start` on resource 'service[datadog-agent]'
    ================================================================================

    Errno::ENOENT
    -------------
    No such file or directory - /sbin/start

```
So I fixed with this. And yes I was quite surprised, I can say what I did worked but I was puzzled to be forced to do that, so there is maybe something I did wrong

Reply:
Replaced with an update on recipe_helpers